api/src/opentrons/protocol_engine/state/commands.py

```
@@ -377,6 +369,44 @@ def handle_action(self, action: Action) -> None:  # noqa: C901
                 elif action.door_state == DoorState.CLOSED:
                     self._state.is_door_blocking = False
 
```

Do these 2 mapping methods need to be separated? Seems like it has the same signature and we can know by the exception type what to raise? Maybe Im missing something

---

I see. for the run we are raising the `UnexpectedProtocolError` and for the finish we are raising `PythonException`